source/validate.cpp

```diff
@@ -332,14 +270,34 @@ void DebugInstructionPass(ValidationState_t& _,
   }
 }
 
+void ProcessIds(ValidationState_t& _, const spv_parsed_instruction_t& inst) {
```

Comment:
These types of checks should ideally be performed by the SsaPass and after the CfgPass. This will allow you to prove dominance. Some of this functionality is already implemented in the current version of the SsaPass.